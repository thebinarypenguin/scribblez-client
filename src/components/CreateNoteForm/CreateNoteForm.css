#CreateNoteForm {
  background-color: #dadada;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 0.5rem;
}
