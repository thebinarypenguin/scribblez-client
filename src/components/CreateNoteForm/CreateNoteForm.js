import React from 'react';

class CreateNoteForm extends React.Component {

  constructor(props) {

    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(ev) {

    ev.preventDefault();

    const body       = ev.target['note-body'].value;
    const visibility = ev.target['note-visibility'].value;

    this.props.onSubmit({ body, visibility });
  }

  render() {

    return (

      <form id="new-note-form" onSubmit={this.handleSubmit}>

        <textarea id="note-body" rows="1" placeholder="New note"></textarea>

        <select id="note-visibility">
          <option value="public">public</option>
          <option value="private">private</option>
        </select>

        <button id="create-note" type="submit">Create</button>

    </form>
    )
  }
}

export default CreateNoteForm;
