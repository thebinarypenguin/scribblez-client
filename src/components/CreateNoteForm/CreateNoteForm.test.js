import React from 'react';
import ReactDOM from 'react-dom';
import CreateNoteForm from './CreateNoteForm';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CreateNoteForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});
