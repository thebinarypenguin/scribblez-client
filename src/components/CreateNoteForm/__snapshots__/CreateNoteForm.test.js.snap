// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Container"
>
  <form
    id="CreateNoteForm"
    onSubmit={[Function]}
  >
    <textarea
      id="note-body"
      placeholder="New note"
      rows="1"
    />
    <select
      id="note-visibility"
    >
      <option
        value="public"
      >
        public
      </option>
      <option
        value="private"
      >
        private
      </option>
    </select>
    <button
      id="create-note"
      type="submit"
    >
      Create
    </button>
  </form>
</div>
`;
